content/en/docs/reference/access-authn-authz/extensible-admission-controllers.md

```diff
@@ -796,4 +796,90 @@ webhooks:
+a user-provided pod.
+
+### Intercepting all versions of an object
+
+It is recommended that admission webhooks should always intercpet all versions of an object by setting `.webhooks[].matchPolicy`
+to `Equivalent`. It is also recommended that admission webhooks should prefer registering for stable versions of resources.
+Failure to intercept all versions of an object can result in admission policies not being enforced for requests in certain
+versions. See [Matching requests: matchPolicy](#matching-requests-matchpolicy) for examples.
+
+### Availability
+
+It is recommended that admission webhooks should should evaluate as quickly as possible, since they add to API request latency.
+It is encouraged to use a small timeout for webhooks. See [Timeouts](#timeouts) for more detail.
+
+It is recommended that admission webhooks should leverage some format of load-balancing, to provide high availability and
+performance benefits. If a webhook is running within the cluster, you can run multiple webhook backends behind a service
+to leverage the load-balancing that service supports.
+
+### Guaranteeing the final state of the object is seen
+
+Admission webhooks that need to guarantee they see the final state of the object in order to enforce policy
+should use a validating admission webhook, since objects can be modified after being seen by mutating webhooks.
+
+For example, a mutating admission webhook is configured to inject a sidecar container with name "foo-sidecar" on every
+`CREATE` pod requests. A validating admisson webhook must be configurated to intercept `CREATE` pod requests, and validate
+that a container with name "foo-sidecar" exists in the to-be-created object.
+
+### Avoiding deadlocks in self-hosted webhooks
+
+A webhook that enacts policies for webhook mission-critical resources might cause deadlocks for its own deployment.
+
+For example, a mutating admission webhook is configured to admit `CREATE` pod requests only if certain label is set in the
+pod. The webhook server runs in a deployment which doesn't set the label. When a node that runs the webhook server pods
+becomes unhealthy, the webhook deployment will try to reschedule the pods to another node. However the requests will
+get rejected by the existing webhook server since the label is unset, and the migration cannot happen.
+
+It is recommended to exclude the namespace where your webhook is running with a [namespaceSelector](#matching-requests-namespaceselector).
+
+### Side effects
+
+It is recommended that admission webhooks should avoid side effects if possible, which means the webhooks operate only on the
+content of the `AdmissionReview` sent to them, and do not make out-of-band changes. The `.webhooks[].sideEffects` field should
+be set to `None` if a webhook doesn't have any side effect.
+
+If side effects are required during the admission evaluation, it is recommended that admission webhooks should not trigger
```

Comment:
```suggestion
If side effects are required during the admission evaluation, they must be suppressed when processing an AdmissionReview object with `dryRun` set to `true`, and the `.webhooks[].sideEffects` field should be set to `NoneOnDryRun`.
```

strengthen this guidance, since only `None` or `NoneOnDryRun` are allowed in v1